File: script-dashboard/src/shared/parse.ts
export function parseParams(params: string): Record<string, unknown> {
  try {
    const p = JSON.parse(params || '{}');
    return typeof p === 'object' && p !== null && !Array.isArray(p) ? p : {};
  } catch {
    return {};
  }
}

export function parseMessageBody(body: string): Record<string, unknown> {
  try {
    const parsed = JSON.parse(body);
    return typeof parsed === 'object' && parsed !== null && !Array.isArray(parsed) ? parsed : { message: body };
  } catch {
    return { message: body };
  }
}

File: script-dashboard/python-server/utils/parse.py
"""JSON parsing utilities."""
import json


def parse_params(params_str: str | None) -> dict:
    """Parse params JSON string, returning empty dict on failure."""
    try:
        parsed = json.loads(params_str) if params_str else {}
        if isinstance(parsed, dict):
            return parsed
        return {}
    except (json.JSONDecodeError, TypeError):
        return {}


def parse_message_body(body) -> dict:
    """Parse a message body string, returning dict on success or the string itself."""
    if body is None:
        return {}
    if isinstance(body, dict):
        return body
    if not isinstance(body, str):
        return {}
    try:
        parsed = json.loads(body)
        if isinstance(parsed, dict):
            return parsed
        return {}
    except (json.JSONDecodeError, TypeError):
        return {}
